### DISCUSSION TITLE
External user of oviewer and QuitSmall

### DISCUSSION THREAD

**Discussion by peloy** [noborus/ov#163] [General]
Hi Noboru,

I want to use the oviewer module for my application but I cannot get QuitSmall to work -- I set it to true but I notice two odd behaviors:

1. The screen is cleared by oviewer.NewRoot() (before calling (*Root).Run() )
2. (*Root).Run() does not quit when the "document" has a handful of lines.

Test program below.

Passing to ov a file with a handful lines, or running "echo test | ./ov -F" yields the behavior I need, but how to achieve that same behavior when my application is an external user of oviewer?

Thanks in advance!

Eloy Paris.-

==========
```
package main

import (
	"bytes"
	"fmt"
	"log"
	"time"

	"github.com/noborus/ov/oviewer"
)

func main() {
	var b bytes.Buffer

	for i := 0; i < 5; i++ {
		b.WriteString(fmt.Sprintf("%d\n", i))
	}

	fmt.Println("Sleeping 3 seconds before calling NewRoot()")
	time.Sleep(3 *time.Second)

	ov, err := oviewer.NewRoot(&b)
	if err != nil {
		log.Fatal(err)
	}
	// oviewer.NewRoot() clears the screen; why?

	ov.QuitSmall = true

	time.Sleep(3 *time.Second)

	if err := ov.Run(); err != nil {
		log.Fatal(err)
	}

	if ov.IsWriteOriginal {
		ov.WriteOriginal()
	}
}
```

**Comment by noborus** [comment]
Hello

In conclusion, it was a bug.
The condition is that docSmall has reached EOF, but NewRoot () has not received the EOF channel.
I fix it as follows:

```diff
diff --git a/oviewer/oviewer.go b/oviewer/oviewer.go
index f35cc91..350e35d 100644
--- a/oviewer/oviewer.go
+++ b/oviewer/oviewer.go
@@ -354,6 +354,9 @@ func NewRoot(read io.Reader) (*Root, error) {
     if err != nil {
        return nil, err
     }
+    go func() {
+       <-m.eofCh
+    }()
     if err := m.ReadAll(read); err != nil {
        return nil, err
     }
```

**Reply by peloy** [reply]
Thanks for the quick response, Noboru; that indeed fixes it! Cheers!

**Reply by noborus** [reply]
Merged. Thank you for your report.
